src/Jaeger/Sampler/RateLimitingSampler.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Jaeger\Sampler;
+
+use Jaeger\Util\RateLimiter;
+
+use const Jaeger\SAMPLER_PARAM_TAG_KEY;
+use const Jaeger\SAMPLER_TYPE_RATE_LIMITING;
+use const Jaeger\SAMPLER_TYPE_TAG_KEY;
+
+class RateLimitingSampler implements SamplerInterface
+{
+    /**
+     * @var RateLimiter
+     */
+    private $rateLimiter;
+
+    /**
+     * A list of the sampler tags.
+     *
+     * @var array
+     */
+    private $tags = [];
+
+    public function __construct($maxTracesPerSecond, RateLimiter $rateLimiter)
+    {
+        $this->tags = [
+            SAMPLER_TYPE_TAG_KEY => SAMPLER_TYPE_RATE_LIMITING,
+            SAMPLER_PARAM_TAG_KEY => $maxTracesPerSecond,
+        ];
+
+        $maxTracesPerNanosecond = $maxTracesPerSecond / 1000000000.0;
+        $this->rateLimiter = $rateLimiter;
+        $this->rateLimiter->initialize($maxTracesPerNanosecond, $maxTracesPerSecond > 1.0 ? 1.0 : $maxTracesPerSecond);
```

Comment:
Just curious about your decision to add the `RateLimiter#initialize` method. Seems odd that you would pass a not fully constructed RateLimiter in this constructor.

It appears that the Go implementation requires you to pass `maxTracesPerSecond` into both the `Sampler` and the `RateLimiter`. Do we gain anything by doing it differently?

Reply:
I needed to do 2 things:

1. is to create instance of it, that will be connected to some Cache and have loaded state from last request - this is related to infrastructure and `RateLimiter` cannot exists with this. Also passing this trough DI makes it easier to test.
2. is to set some custom rate limiting params. These can be changed even by runtime. I decided to set them in `RateLimitingSampler`, because it applies to those params some Sampler calculations, before setting proper `RateLimiter` configurations. - this relates more to business logic of `RateLimitingSampler` itself.

I think main gain is quite nice separation of infra and business logic.